### DISCUSSION TITLE
using-source-generators-with-blazor-in-dotnet-6/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#363] [Announcements]
# using-source-generators-with-blazor-in-dotnet-6/

In this post I describe a problem where upgrading my Blazor app to .NET 6 broke my source generators that relied on Blazor components.

https://andrewlock.net/using-source-generators-with-blazor-in-dotnet-6/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at March 29 2022, 08:38, in reply to [torrey.garland](https://disqus.com/by/torreygarland/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/363#discussioncomment-6059898)</em>

---

<p>&gt; Would placing the route attributes on the code behind class work</p><p>Yeah, it probably would! It's not idiomatic Razor then unfortunately...</p><p>&gt;has microsoft fixed this issue yet and just not documented it</p><p>Not as far as I know 🤔 Was it a .NET 5 or .NET 6 project? 🤔</p>

[a comment by an automated account is not shown]

**Comment by olizarevichroman** [comment]
Hi, I've tried to do the same passing UseRazorSourceGenerator=false, but it seems like generated ".cs" files are not passed into a source generator. Do you have any ideas?

Thanks!
